soc/nordic/common/uicr/gen_uicr/CMakeLists.txt

```diff
@@ -0,0 +1,109 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The code in this CMakeLists.txt constructs the arguments for gen_uicr.py
+# and creates a flashable zephyr.hex file containing UICR data (no C code compiled)
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+# Instead of adding all of Zephyr we add just the subset that is
+# required to generate uicr.hex.
+#
+# The generation of uicr.hex is configured by this image, so we
+# include modules from zephyr_default up until kconfig.
+
+find_package(Zephyr
+  COMPONENTS zephyr_default:kconfig
+  REQUIRED HINTS $ENV{ZEPHYR_BASE}
+  )
+
+project(uicr)
+
+# Function to compute partition absolute address and size from devicetree
+function(compute_partition_address_and_size partition_nodelabel output_address_var output_size_var)
+  # Get partition path from devicetree
+  dt_nodelabel(partition_path NODELABEL ${partition_nodelabel})
+  if(NOT DEFINED partition_path)
+    message(FATAL_ERROR "${partition_nodelabel} not found in devicetree")
+  endif()
+
+  # Extract offset and size from devicetree
+  dt_reg_addr(partition_offset PATH ${partition_path})
+  dt_reg_size(partition_size PATH ${partition_path})
+
+  if(NOT DEFINED partition_offset OR NOT DEFINED partition_size)
+    message(FATAL_ERROR "Failed to read ${partition_nodelabel} addresses from DT")
+  endif()
+
+  # Calculate absolute partition address
+  math(EXPR partition_address "${CONFIG_FLASH_BASE_ADDRESS} + ${partition_offset}" OUTPUT_FORMAT HEXADECIMAL)
+
+  # Set output variables in parent scope
+  set(${output_address_var} ${partition_address} PARENT_SCOPE)
+  set(${output_size_var} ${partition_size} PARENT_SCOPE)
+endfunction()
+
+# Use CMAKE_VERBOSE_MAKEFILE to silence an unused-variable warning.
+if(CMAKE_VERBOSE_MAKEFILE)
+endif()
+
+set(periphconf_args)
+set(periphconf_elfs)
+set(merged_hex_file ${APPLICATION_BINARY_DIR}/zephyr/zephyr.hex)
```

Comment:
use the Kconfig for this for the bin name

Reply:
fixed

I couldn't find neither a variable or a Kconfig that contained the hex suffix unfortunately.